### PR TITLE
BUILD: Add debian userspace packaging

### PR DESCRIPTION
**Pull request by tvegas1**
Add debian userspace packaging from @tzafrir-mellanox, and trigger in CI.

### CODE REVIEW

**Comment by tvegas1** [comment]
```
dpkg --contents libxpmem_2.7.3_amd64.deb
...
-rw-r--r-- root/root     14480 2023-02-13 15:50 ./usr/lib/x86_64-linux-gnu/libxpmem.so.0.0.0
...
dpkg --contents libxpmem-dev_2.7.3_amd64.deb
...
-rw-r--r-- root/root      6062 2023-02-13 15:50 ./usr/include/xpmem.h
drwxr-xr-x root/root         0 2023-02-13 15:50 ./usr/lib/
drwxr-xr-x root/root         0 2023-02-13 15:50 ./usr/lib/x86_64-linux-gnu/
drwxr-xr-x root/root         0 2023-02-13 15:50 ./usr/lib/x86_64-linux-gnu/pkgconfig/
-rw-r--r-- root/root       197 2023-02-13 15:50 ./usr/lib/x86_64-linux-gnu/pkgconfig/cray-xpmem.pc
drwxr-xr-x root/root         0 2023-02-13 15:50 ./usr/share/
drwxr-xr-x root/root         0 2023-02-13 15:50 ./usr/share/doc/
drwxr-xr-x root/root         0 2023-02-13 15:50 ./usr/share/doc/libxpmem-dev/
-rw-r--r-- root/root       147 2023-02-13 15:50 ./usr/share/doc/libxpmem-dev/changelog.gz
-rw-r--r-- root/root      1102 2023-02-13 15:50 ./usr/share/doc/libxpmem-dev/copyright
lrwxrwxrwx root/root         0 2023-02-13 15:50 ./usr/lib/x86_64-linux-gnu/libxpmem.so -> libxpmem.so.0.0.0
...
dpkg --contents libxpmem-dbgsym_2.7.3_amd64.ddeb
...
```

File: debian/control [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Source: xpmem-lib
+Priority: optional
+Maintainer: Tzafrir Cohen <nvidia@cohens.org.il>
```
Comment (yosefe) [inline debian/control]: Shall we update email to ucx team?

Comment (tvegas1) [reply debian/control]: fixed

File: ci/azure-pipelines.yml [hunk 2]
```diff
@@ -170,6 +170,26 @@ stages:
...
+              ./configure --disable-kernel-module
+              make dist-gzip
+              tarball="$(ls xpmem-*.tar.gz)"
+              tar zxf "$tarball"
+              cd "${tarball%.tar.gz}"
+              dpkg-buildpackage --build=binary -uc
+              cd -
+              dpkg --contents libxpmem_*.deb
+              dpkg --contents libxpmem-dev_*.deb
+              dpkg --contents libxpmem-dbgsym_*.ddeb
```
Comment (brminich) [inline ci/azure-pipelines.yml:189]: is it worth to check the content in some way? (i.e. parse the output)

Comment (tvegas1) [reply ci/azure-pipelines.yml:189]: pushed to try to improve tests

**Review by brminich — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Pull request merged**